doc/source/area-detector.rst

```diff
@@ -269,6 +269,168 @@ The vendor specific details are embedded in the cams
    :parts: 1
 
+Custom Devices
+==============
+For custom hardware based on area-detector it may be necesary to add a custom
+device class (for custom plugins see section below). The new class should
+inherit from ``ophyd.areadetector.base.ADbase`` and should have the following
+PV structure:
+
+.. code-block:: python
+
+    PV = Areadetector_device_PV_prefix:(Plugin_suffix or attribute_suffix)
+
+As an example, for the builtin areadetector 'stats' class this looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:Stats
+
+And for the builtin areadetector 'color mode' attribute it looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:ColorMode_RBV
+
+where ``Areadetector_device_PV_suffix`` is the PV name for the Area detector
+device, ``plugin_suffix = Stats`` is the 'stats' Plugin suffix and
+``attribute_suffix = ColorMode_RBV`` is the'color mode' attribute suffix.
+
+In order to create the class then the following code is required (where ``XXX``
+is the name of the device):
+
+.. code-block:: python
+
+    from ophyd.areadetector.base import ADComponent as ADCpt, ad_group,
+                                        EpicsSignalWithRBV
+    from ophyd.signal import EpicsSignal, EpicsSignalRO
+    from ophyd.device import DynamicDeviceComponent as DDCpt
+    from ophyd.detectors import DetectorBase
+
+    Class XXX('optional trigger mixin, 'DetectorBase):
+        '''An areadetector device class for ......'''
+        _suffix_re = '"Areadetector_suffix"\d:'
+
+        # ADD ATTRIBUTES AS COMPONENTS HERE USING THE SYNTAX
+        attribute_name = Cpt(Type, attribute_suffix)
+            # where 'Type' is EpicsSignal, EpicsSignalRO, EpicsSignalWithRBV,..
+
+        # ADD ATTRIBUTE GROUPS AS COMPONENTS USING THE SYNTAX
+        group_name = DDCpt(ad_group(Type,
+                                    (attribute_1_name, attribute_1_suffix),
+                                    (attribute_2_name, attribute_2_suffix),
+                                    ...................
+                                    (attribute_n_name, attribute_n_suffix))
+
+        # ADD ATTRIBUTE PLUGINS AS COMPONENTS USING THE SYNTAX
+        plugin_name = Cpt(PluginClass, suffix = Plugin_suffix+':')
+
+
+.. note::
+    1. DetectorBase can be swapped out for any other Areadetector Device
+    class that inherits from DetectorBase. Also an optional trigger_mixin
+    class can be specifed (trigger_mixin classes must be listed first).
+
+    2. Component (Cpt) can be replaced with ADComponent(ADCpt) if this is a
```

Comment:
I would rather steer people away from this entirely, and avoid importing it in the examples above (since it isn't used). Suggested wording:

```rst
In the ophyd source code, you may see ``ADComponent`` used. Functionally, this is interchangeable with 
an ordinary ophyd ``Component`` (imported as ``Cpt`` above); it just adds extra machinery for generated 
a docstring based on a scrape of the HTML of the official AreaDetector documentation. For custom 
extensions such as we are addressing here, it is not generally applicable.
```

Reply:
made this change